src/uri.rs

```diff
@@ -0,0 +1,261 @@
+#![allow(dead_code)]
+
+use reqwest;
+
+use serde_derive::Deserialize;
+
+// these item structs should be namespaced into the v2 api.
+#[derive(Deserialize, Debug)]
+struct ItemAttribute {
+    attribute: String,
+    modifier: u32,
+}
+
+#[derive(Deserialize, Debug)]
+struct ItemDetails {
+    #[serde(rename = "type")]
+    c_type: String,
+    weight_class: String,
+    defense: u32,
+    // whats the type of infusions slots item?
+    //infusion_slots:
+    //infix_upgrade : HashMap<String,
+    // infix upgrade is of format { "id" : u32, "attributes" : Vec<ItemAttribute> }
+    // bit un certain how to tell serdea the values are of different types https://github.com/serde-rs/json/issues/144
+    attribute_adjustment: f64,
+    suffix_item_id: u64,
+    secondary_suffix_item_id: Option<String>,
+}
+
+#[derive(Deserialize, Debug)]
+struct Item {
+    name: String,
+    // maybe should be an option?
+    description: String,
+    #[serde(rename = "type")]
+    item_type: String,
+    id: u32,
+    level: u32,
+    rarity: String,
+    default_skin: u64,
+    game_types: Vec<String>,
+    flags: Vec<String>,
+    // not certain what the vec type should be.
+    restrictions: Vec<String>,
+    chat_link: String,
+    icon: String,
+    details: ItemDetails,
+}
+
+// should be usize?
```

Comment:
Using usize for values coming from a web API is iffy to me.  Introduces different platforms treating the same data slightly differently for no real compelling reason.

Reply:
ah sure, this was here as a question of not really know what the range of values from the api would be returned as.